From: Debian Rust Maintainers <pkg-rust-maintainers@alioth-lists.debian.net>
Date: Mon, 6 May 2024 10:25:32 +0200
Subject: d-0020-remove-windows-dependencies

use something like

 find src compiler library -iname Cargo.toml -exec grep -H -n -e 'windows-sys' -e 'winapi' -e 'ntapi' -e 'wincon' -e 'winreg' -e 'windows' {} \;

to find and eliminate dependencies on windows-only crates when rebasing.

windows-bindgen and windows-metadata should not be removed, they are needed for
the build and don't pull in windows-sys and friends.

Forwarded: not-needed

===================================================================
---
 compiler/rustc_codegen_ssa/Cargo.toml              |  4 ----
 compiler/rustc_data_structures/Cargo.toml          | 10 ---------
 compiler/rustc_driver_impl/Cargo.toml              |  6 -----
 compiler/rustc_errors/Cargo.toml                   |  8 -------
 compiler/rustc_session/Cargo.toml                  |  7 ------
 library/backtrace/Cargo.toml                       |  3 ---
 library/backtrace/crates/as-if-std/Cargo.toml      |  3 ---
 src/bootstrap/Cargo.toml                           | 15 -------------
 src/tools/cargo/Cargo.toml                         | 26 ++++------------------
 .../cargo/crates/cargo-test-support/Cargo.toml     |  3 ---
 src/tools/cargo/crates/cargo-util/Cargo.toml       |  7 ------
 src/tools/cargo/crates/home/Cargo.toml             |  3 ---
 .../cargo/credential/cargo-credential/Cargo.toml   |  3 ---
 src/tools/cargo/src/cargo/util/auth/mod.rs         |  5 -----
 src/tools/compiletest/Cargo.toml                   | 10 ---------
 src/tools/rust-analyzer/crates/profile/Cargo.toml  |  6 -----
 .../rust-analyzer/crates/rust-analyzer/Cargo.toml  |  6 -----
 src/tools/rust-analyzer/crates/stdx/Cargo.toml     |  4 ----
 src/tools/rustc-perf/collector/Cargo.toml          |  4 ----
 .../compile-benchmarks/cargo-0.60.0/Cargo.toml     | 24 --------------------
 .../style-servo/components/gfx/Cargo.toml          |  4 ----
 .../style-servo/components/style/Cargo.toml        |  3 ---
 .../native-tls-0.1.5/Cargo.toml                    |  4 +---
 .../example-compositor/compositor/Cargo.toml       |  3 ---
 .../webrender-2022/webrender/Cargo.toml            |  3 ---
 .../webrender-2022/wrench/Cargo.toml               |  4 ----
 26 files changed, 5 insertions(+), 173 deletions(-)

diff --git a/compiler/rustc_codegen_ssa/Cargo.toml b/compiler/rustc_codegen_ssa/Cargo.toml
index 97eebffd..1ddf0fc 100644
--- a/compiler/rustc_codegen_ssa/Cargo.toml
+++ b/compiler/rustc_codegen_ssa/Cargo.toml
@@ -56,7 +56,3 @@ libc = "0.2.50"
 version = "0.36.2"
 default-features = false
 features = ["read_core", "elf", "macho", "pe", "xcoff", "unaligned", "archive", "write", "wasm"]
-
-[target.'cfg(windows)'.dependencies.windows]
-version = "0.59.0"
-features = ["Win32_Globalization"]
diff --git a/compiler/rustc_data_structures/Cargo.toml b/compiler/rustc_data_structures/Cargo.toml
index fcaf275..9685698 100644
--- a/compiler/rustc_data_structures/Cargo.toml
+++ b/compiler/rustc_data_structures/Cargo.toml
@@ -37,16 +37,6 @@ features = ["nightly"] # for may_dangle
 [dependencies.parking_lot]
 version = "0.12"
 
-[target.'cfg(windows)'.dependencies.windows]
-version = "0.59.0"
-features = [
-    "Win32_Foundation",
-    "Win32_Storage_FileSystem",
-    "Win32_System_IO",
-    "Win32_System_ProcessStatus",
-    "Win32_System_Threading",
-]
-
 [target.'cfg(unix)'.dependencies]
 # tidy-alphabetical-start
 libc = "0.2"
diff --git a/compiler/rustc_driver_impl/Cargo.toml b/compiler/rustc_driver_impl/Cargo.toml
index de64335..1322da3 100644
--- a/compiler/rustc_driver_impl/Cargo.toml
+++ b/compiler/rustc_driver_impl/Cargo.toml
@@ -59,12 +59,6 @@ tracing = { version = "0.1.35" }
 libc = "0.2"
 # tidy-alphabetical-end
 
-[target.'cfg(windows)'.dependencies.windows]
-version = "0.59.0"
-features = [
-    "Win32_System_Diagnostics_Debug",
-]
-
 [target.'cfg(not(target_family = "wasm"))'.dependencies]
 # tidy-alphabetical-start
 ctrlc = "3.4.4"
diff --git a/compiler/rustc_errors/Cargo.toml b/compiler/rustc_errors/Cargo.toml
index b11793c..aa1ec11 100644
--- a/compiler/rustc_errors/Cargo.toml
+++ b/compiler/rustc_errors/Cargo.toml
@@ -31,11 +31,3 @@ termcolor = "1.2.0"
 termize = "0.1.1"
 tracing = "0.1"
 # tidy-alphabetical-end
-
-[target.'cfg(windows)'.dependencies.windows]
-version = "0.59.0"
-features = [
-    "Win32_Foundation",
-    "Win32_Security",
-    "Win32_System_Threading",
-]
diff --git a/compiler/rustc_session/Cargo.toml b/compiler/rustc_session/Cargo.toml
index a087725..32744a1 100644
--- a/compiler/rustc_session/Cargo.toml
+++ b/compiler/rustc_session/Cargo.toml
@@ -30,10 +30,3 @@ tracing = "0.1"
 # tidy-alphabetical-start
 libc = "0.2"
 # tidy-alphabetical-end
-
-[target.'cfg(windows)'.dependencies.windows]
-version = "0.59.0"
-features = [
-    "Win32_Foundation",
-    "Win32_System_LibraryLoader",
-]
diff --git a/library/backtrace/Cargo.toml b/library/backtrace/Cargo.toml
index c2d4f15..2bc9ae7 100644
--- a/library/backtrace/Cargo.toml
+++ b/library/backtrace/Cargo.toml
@@ -38,9 +38,6 @@ cpp_demangle = { default-features = false, version = "0.4.0", optional = true, f
   "alloc",
 ] }
 
-[target.'cfg(windows)'.dependencies]
-windows-targets = "0.52.6"
-
 [target.'cfg(not(all(windows, target_env = "msvc", not(target_vendor = "uwp"))))'.dependencies]
 miniz_oxide = { version = "0.8", default-features = false }
 ruzstd = { version = "0.7.3", default-features = false, optional = true }
diff --git a/library/backtrace/crates/as-if-std/Cargo.toml b/library/backtrace/crates/as-if-std/Cargo.toml
index 092905f..bfb0700 100644
--- a/library/backtrace/crates/as-if-std/Cargo.toml
+++ b/library/backtrace/crates/as-if-std/Cargo.toml
@@ -27,9 +27,6 @@ default-features = false
 optional = true
 features = ['read_core', 'elf', 'macho', 'pe', 'xcoff', 'unaligned', 'archive']
 
-[target.'cfg(windows)'.dependencies]
-windows-targets = "0.52.6"
-
 [features]
 default = ['backtrace']
 backtrace = ['addr2line', 'miniz_oxide', 'object', 'ruzstd']
diff --git a/src/bootstrap/Cargo.toml b/src/bootstrap/Cargo.toml
index 23aa87a..79330ae 100644
--- a/src/bootstrap/Cargo.toml
+++ b/src/bootstrap/Cargo.toml
@@ -66,21 +66,6 @@ tracing-chrome = { version = "0.7", optional = true }
 tracing-subscriber = { version = "0.3", optional = true, features = ["env-filter", "fmt", "registry", "std"] }
 tracing-tree = { version = "0.4.0", optional = true }
 
-[target.'cfg(windows)'.dependencies.junction]
-version = "1.0.0"
-
-[target.'cfg(windows)'.dependencies.windows]
-version = "0.57"
-features = [
-    "Win32_Foundation",
-    "Win32_Security",
-    "Win32_System_Diagnostics_Debug",
-    "Win32_System_JobObjects",
-    "Win32_System_ProcessStatus",
-    "Win32_System_Threading",
-    "Win32_System_Time",
-]
-
 [dev-dependencies]
 pretty_assertions = "1.4"
 
diff --git a/src/tools/cargo/Cargo.toml b/src/tools/cargo/Cargo.toml
index adac1c7..44fccab 100644
--- a/src/tools/cargo/Cargo.toml
+++ b/src/tools/cargo/Cargo.toml
@@ -2,7 +2,9 @@
 resolver = "2"
 members = [
   "crates/*",
-  "credential/*",
+  "credential/cargo-credential",
+  "credential/cargo-credential-1password",
+  "credential/cargo-credential-libsecret",
   "benches/benchsuite",
   "benches/capture",
 ]
@@ -28,8 +30,6 @@ build-rs = { version = "0.3.0", path = "crates/build-rs" }
 cargo = { path = "" }
 cargo-credential = { version = "0.4.2", path = "credential/cargo-credential" }
 cargo-credential-libsecret = { version = "0.4.13", path = "credential/cargo-credential-libsecret" }
-cargo-credential-macos-keychain = { version = "0.4.13", path = "credential/cargo-credential-macos-keychain" }
-cargo-credential-wincred = { version = "0.4.13", path = "credential/cargo-credential-wincred" }
 cargo-platform = { path = "crates/cargo-platform", version = "0.2.0" }
 cargo-test-macro = { version = "0.4.2", path = "crates/cargo-test-macro" }
 cargo-test-support = { version = "0.7.1", path = "crates/cargo-test-support" }
@@ -115,7 +115,6 @@ unicode-xid = "0.2.6"
 url = "2.5.4"
 varisat = "0.2.2"
 walkdir = "2.5.0"
-windows-sys = "0.59"
 
 [workspace.lints.rust]
 rust_2018_idioms = "warn" # TODO: could this be removed?
@@ -157,6 +156,7 @@ anyhow.workspace = true
 base64.workspace = true
 blake3.workspace = true
 cargo-credential.workspace = true
+cargo-credential-libsecret.workspace = true
 cargo-platform.workspace = true
 cargo-util-schemas.workspace = true
 cargo-util.workspace = true
@@ -227,27 +227,9 @@ libc.workspace = true
 [target.'cfg(target_os = "linux")'.dependencies]
 cargo-credential-libsecret.workspace = true
 
-[target.'cfg(target_os = "macos")'.dependencies]
-cargo-credential-macos-keychain.workspace = true
-
 [target.'cfg(not(windows))'.dependencies]
 openssl = { workspace = true, optional = true }
 
-[target.'cfg(windows)'.dependencies]
-cargo-credential-wincred.workspace = true
-
-[target.'cfg(windows)'.dependencies.windows-sys]
-workspace = true
-features = [
-  "Win32_Foundation",
-  "Win32_Security",
-  "Win32_Storage_FileSystem",
-  "Win32_System_IO",
-  "Win32_System_Console",
-  "Win32_System_JobObjects",
-  "Win32_System_Threading",
-]
-
 [dev-dependencies]
 annotate-snippets = { workspace = true, features = ["testing-colors"] }
 cargo-test-support.workspace = true
diff --git a/src/tools/cargo/crates/cargo-test-support/Cargo.toml b/src/tools/cargo/crates/cargo-test-support/Cargo.toml
index a61934a..13f9dfb 100644
--- a/src/tools/cargo/crates/cargo-test-support/Cargo.toml
+++ b/src/tools/cargo/crates/cargo-test-support/Cargo.toml
@@ -31,8 +31,5 @@ toml.workspace = true
 url.workspace = true
 walkdir.workspace = true
 
-[target.'cfg(windows)'.dependencies]
-windows-sys = { workspace = true, features = ["Win32_Storage_FileSystem"] }
-
 [lints]
 workspace = true
diff --git a/src/tools/cargo/crates/cargo-util/Cargo.toml b/src/tools/cargo/crates/cargo-util/Cargo.toml
index fde0193..efe776a 100644
--- a/src/tools/cargo/crates/cargo-util/Cargo.toml
+++ b/src/tools/cargo/crates/cargo-util/Cargo.toml
@@ -21,15 +21,8 @@ tempfile.workspace = true
 tracing.workspace = true
 walkdir.workspace = true
 
-[target.'cfg(target_os = "macos")'.dependencies]
-core-foundation.workspace = true
-
 [target.'cfg(unix)'.dependencies]
 libc.workspace = true
 
-[target.'cfg(windows)'.dependencies]
-miow.workspace = true
-windows-sys = { workspace = true, features = ["Win32_Storage_FileSystem", "Win32_Foundation", "Win32_System_Console"] }
-
 [lints]
 workspace = true
diff --git a/src/tools/cargo/crates/home/Cargo.toml b/src/tools/cargo/crates/home/Cargo.toml
index 747316f..5a313a9 100644
--- a/src/tools/cargo/crates/home/Cargo.toml
+++ b/src/tools/cargo/crates/home/Cargo.toml
@@ -17,8 +17,5 @@ homepage.workspace = true
 repository.workspace = true
 description = "Shared definitions of home directories."
 
-[target.'cfg(windows)'.dependencies]
-windows-sys = { workspace = true, features = ["Win32_Foundation", "Win32_UI_Shell", "Win32_System_Com"] }
-
 [lints]
 workspace = true
diff --git a/src/tools/cargo/credential/cargo-credential/Cargo.toml b/src/tools/cargo/credential/cargo-credential/Cargo.toml
index 713fa41..5017448 100644
--- a/src/tools/cargo/credential/cargo-credential/Cargo.toml
+++ b/src/tools/cargo/credential/cargo-credential/Cargo.toml
@@ -18,9 +18,6 @@ time.workspace = true
 [target.'cfg(unix)'.dependencies]
 libc.workspace = true
 
-[target.'cfg(windows)'.dependencies]
-windows-sys = { workspace = true, features = ["Win32_System_Console", "Win32_Foundation"] }
-
 [dev-dependencies]
 snapbox = { workspace = true, features = ["examples"] }
 
diff --git a/src/tools/cargo/src/cargo/util/auth/mod.rs b/src/tools/cargo/src/cargo/util/auth/mod.rs
index 2dc66f5..92d240b 100644
--- a/src/tools/cargo/src/cargo/util/auth/mod.rs
+++ b/src/tools/cargo/src/cargo/util/auth/mod.rs
@@ -540,11 +540,6 @@ fn credential_action(
             }
             "cargo:paseto" => bail!("cargo:paseto requires -Zasymmetric-token"),
             "cargo:token-from-stdout" => Box::new(BasicProcessCredential {}),
-            #[cfg(windows)]
-            "cargo:wincred" => Box::new(cargo_credential_wincred::WindowsCredential {}),
-            #[cfg(target_os = "macos")]
-            "cargo:macos-keychain" => Box::new(cargo_credential_macos_keychain::MacKeychain {}),
-            #[cfg(target_os = "linux")]
             "cargo:libsecret" => Box::new(cargo_credential_libsecret::LibSecretCredential {}),
             name if BUILT_IN_PROVIDERS.contains(&name) => {
                 Box::new(cargo_credential::UnsupportedCredential {})
diff --git a/src/tools/compiletest/Cargo.toml b/src/tools/compiletest/Cargo.toml
index 4f8e475..7303972 100644
--- a/src/tools/compiletest/Cargo.toml
+++ b/src/tools/compiletest/Cargo.toml
@@ -29,13 +29,3 @@ home = "0.5.5"
 
 [target.'cfg(unix)'.dependencies]
 libc = "0.2"
-
-[target.'cfg(windows)'.dependencies]
-miow = "0.6"
-
-[target.'cfg(windows)'.dependencies.windows]
-version = "0.59.0"
-features = [
-    "Win32_Foundation",
-    "Win32_System_Diagnostics_Debug",
-]
diff --git a/src/tools/rust-analyzer/crates/profile/Cargo.toml b/src/tools/rust-analyzer/crates/profile/Cargo.toml
index ad8b6c9..6d28678 100644
--- a/src/tools/rust-analyzer/crates/profile/Cargo.toml
+++ b/src/tools/rust-analyzer/crates/profile/Cargo.toml
@@ -18,12 +18,6 @@ libc.workspace = true
 [target.'cfg(all(target_os = "linux", not(target_env = "ohos")))'.dependencies]
 perf-event = "=0.4.7"
 
-[target.'cfg(windows)'.dependencies]
-windows-sys = { version = "0.59", features = [
-    "Win32_System_Threading",
-    "Win32_System_ProcessStatus",
-] }
-
 [features]
 cpu_profiler = []
 
diff --git a/src/tools/rust-analyzer/crates/rust-analyzer/Cargo.toml b/src/tools/rust-analyzer/crates/rust-analyzer/Cargo.toml
index 7a80a14..f3f9afe 100644
--- a/src/tools/rust-analyzer/crates/rust-analyzer/Cargo.toml
+++ b/src/tools/rust-analyzer/crates/rust-analyzer/Cargo.toml
@@ -73,12 +73,6 @@ vfs-notify.workspace = true
 vfs.workspace = true
 paths.workspace = true
 
-[target.'cfg(windows)'.dependencies]
-windows-sys = { version = "0.59", features = [
-  "Win32_System_Diagnostics_Debug",
-  "Win32_System_Threading",
-] }
-
 [dev-dependencies]
 expect-test = "1.4.0"
 xshell.workspace = true
diff --git a/src/tools/rust-analyzer/crates/stdx/Cargo.toml b/src/tools/rust-analyzer/crates/stdx/Cargo.toml
index 3727d0c..10a86b9 100644
--- a/src/tools/rust-analyzer/crates/stdx/Cargo.toml
+++ b/src/tools/rust-analyzer/crates/stdx/Cargo.toml
@@ -20,10 +20,6 @@ itertools.workspace = true
 tracing.workspace = true
 # Think twice before adding anything here
 
-[target.'cfg(windows)'.dependencies]
-miow = "0.6.0"
-windows-sys = { version = "0.59", features = ["Win32_Foundation"] }
-
 [features]
 # Uncomment to enable for the whole crate graph
 # default = [ "backtrace" ]
diff --git a/src/tools/rustc-perf/collector/Cargo.toml b/src/tools/rustc-perf/collector/Cargo.toml
index d34b023..3d8a5fa 100644
--- a/src/tools/rustc-perf/collector/Cargo.toml
+++ b/src/tools/rustc-perf/collector/Cargo.toml
@@ -46,10 +46,6 @@ inquire = "0.7.5"
 benchlib = { path = "benchlib" }
 database = { path = "../database" }
 
-[target.'cfg(windows)'.dependencies]
-miow = "0.3"
-windows-sys = { version = "0.45.0", features = ["Win32_Foundation"] }
-
 [features]
 # Enable more precise Cachegrind profiles for runtime benchmarks.
 # Requires a recent Valgrind to be installed.
diff --git a/src/tools/rustc-perf/collector/compile-benchmarks/cargo-0.60.0/Cargo.toml b/src/tools/rustc-perf/collector/compile-benchmarks/cargo-0.60.0/Cargo.toml
index 12ee1ee..a89f82f 100644
--- a/src/tools/rustc-perf/collector/compile-benchmarks/cargo-0.60.0/Cargo.toml
+++ b/src/tools/rustc-perf/collector/compile-benchmarks/cargo-0.60.0/Cargo.toml
@@ -209,28 +209,4 @@ deny-warnings = []
 pretty-env-logger = ["pretty_env_logger"]
 vendored-openssl = ["openssl/vendored"]
 
-[target."cfg(windows)".dependencies.fwdansi]
-version = "1.1.0"
-
-[target."cfg(windows)".dependencies.winapi]
-version = "0.3"
-features = [
-    "basetsd",
-    "handleapi",
-    "jobapi",
-    "jobapi2",
-    "memoryapi",
-    "minwindef",
-    "ntdef",
-    "ntstatus",
-    "processenv",
-    "processthreadsapi",
-    "psapi",
-    "synchapi",
-    "winerror",
-    "winbase",
-    "wincon",
-    "winnt",
-]
-
 [workspace]
diff --git a/src/tools/rustc-perf/collector/compile-benchmarks/style-servo/components/gfx/Cargo.toml b/src/tools/rustc-perf/collector/compile-benchmarks/style-servo/components/gfx/Cargo.toml
index 3d7aea1..57feefe 100644
--- a/src/tools/rustc-perf/collector/compile-benchmarks/style-servo/components/gfx/Cargo.toml
+++ b/src/tools/rustc-perf/collector/compile-benchmarks/style-servo/components/gfx/Cargo.toml
@@ -59,7 +59,3 @@ xml5ever = {version = "0.10"}
 
 [target.'cfg(any(target_feature = "sse2", target_feature = "neon"))'.dependencies]
 simd = "0.2.0"
-
-[target.'cfg(target_os = "windows")'.dependencies]
-dwrote = "0.4"
-truetype = "0.26"
diff --git a/src/tools/rustc-perf/collector/compile-benchmarks/style-servo/components/style/Cargo.toml b/src/tools/rustc-perf/collector/compile-benchmarks/style-servo/components/style/Cargo.toml
index 23130d0..5f33889 100644
--- a/src/tools/rustc-perf/collector/compile-benchmarks/style-servo/components/style/Cargo.toml
+++ b/src/tools/rustc-perf/collector/compile-benchmarks/style-servo/components/style/Cargo.toml
@@ -77,9 +77,6 @@ time = "0.1"
 unicode-bidi = "0.3"
 unicode-segmentation = "1.0"
 
-[target.'cfg(windows)'.dependencies]
-kernel32-sys = "0.2"
-
 [build-dependencies]
 lazy_static = "0.2"
 log = "0.3"
diff --git a/src/tools/rustc-perf/collector/compile-benchmarks/tokio-webpush-simple/native-tls-0.1.5/Cargo.toml b/src/tools/rustc-perf/collector/compile-benchmarks/tokio-webpush-simple/native-tls-0.1.5/Cargo.toml
index 38bd630e..6abd17c 100644
--- a/src/tools/rustc-perf/collector/compile-benchmarks/tokio-webpush-simple/native-tls-0.1.5/Cargo.toml
+++ b/src/tools/rustc-perf/collector/compile-benchmarks/tokio-webpush-simple/native-tls-0.1.5/Cargo.toml
@@ -33,7 +33,5 @@ version = "0.1.15"
 
 [target."cfg(any(target_os = \"macos\", target_os = \"ios\"))".dependencies.tempdir]
 version = "0.3"
-[target."cfg(target_os = \"windows\")".dependencies.schannel]
-version = "0.1.7"
 
-[workspace]
\ No newline at end of file
+[workspace]
diff --git a/src/tools/rustc-perf/collector/compile-benchmarks/webrender-2022/example-compositor/compositor/Cargo.toml b/src/tools/rustc-perf/collector/compile-benchmarks/webrender-2022/example-compositor/compositor/Cargo.toml
index 4202332..2d35787 100644
--- a/src/tools/rustc-perf/collector/compile-benchmarks/webrender-2022/example-compositor/compositor/Cargo.toml
+++ b/src/tools/rustc-perf/collector/compile-benchmarks/webrender-2022/example-compositor/compositor/Cargo.toml
@@ -9,8 +9,5 @@ license = "MPL-2.0"
 webrender = { path = "../../webrender" }
 gleam = "0.13.1"
 
-[target.'cfg(windows)'.dependencies]
-compositor-windows = { path = "../compositor-windows" }
-
 [target.'cfg(target_os = "linux")'.dependencies]
 compositor-wayland = { path = "../compositor-wayland" }
diff --git a/src/tools/rustc-perf/collector/compile-benchmarks/webrender-2022/webrender/Cargo.toml b/src/tools/rustc-perf/collector/compile-benchmarks/webrender-2022/webrender/Cargo.toml
index ee727d4..ddc3c14 100644
--- a/src/tools/rustc-perf/collector/compile-benchmarks/webrender-2022/webrender/Cargo.toml
+++ b/src/tools/rustc-perf/collector/compile-benchmarks/webrender-2022/webrender/Cargo.toml
@@ -61,9 +61,6 @@ rand = "0.4"
 freetype = { version = "0.7", default-features = false }
 libc = "0.2"
 
-[target.'cfg(target_os = "windows")'.dependencies]
-dwrote = "0.11"
-
 [target.'cfg(target_os = "macos")'.dependencies]
 core-foundation = "0.9.2"
 core-graphics = "0.22.3"
diff --git a/src/tools/rustc-perf/collector/compile-benchmarks/webrender-2022/wrench/Cargo.toml b/src/tools/rustc-perf/collector/compile-benchmarks/webrender-2022/wrench/Cargo.toml
index 04101b9..9b6e1e5 100644
--- a/src/tools/rustc-perf/collector/compile-benchmarks/webrender-2022/wrench/Cargo.toml
+++ b/src/tools/rustc-perf/collector/compile-benchmarks/webrender-2022/wrench/Cargo.toml
@@ -42,10 +42,6 @@ default = [ "env_logger" ]
 headless = [ "osmesa-sys", "osmesa-src" ]
 software = [ "swgl" ]
 
-[target.'cfg(target_os = "windows")'.dependencies]
-dwrote = "0.11"
-mozangle = { version = "0.3.2", features = ["egl"] }
-
 [target.'cfg(all(unix, not(target_os = "android")))'.dependencies]
 font-loader = "0.11"
 
